sycl-fusion/passes/kernel-info/SYCLKernelInfo.cpp

```diff
@@ -34,6 +32,12 @@ static constexpr std::array<llvm::StringLiteral, NumParameterKinds>
 };
 static constexpr llvm::StringLiteral InvalidParameterKindString{"Invalid"};
 
+template <typename T> static T getUInt(const MDOperand &Operand) {
```

Comment:
> I guess he's using this to avoid avoid having to unwrap the error

Yup, exactly what happened :wink: I originally used `metadataToUInt`, but I found it does too little to justify the heavyweight error handling. My thinking was that just `assert`ing is fine here, because this is the code path for loading a mockup `SYCLKernelInfo` from metadata in lit tests.

Reply:
Yeah, SGTM, but can I propose to do this in a follow-up PR? `KernelTranslatation.cpp` and `SYCLKernelFusion.cpp` have similar local helpers, so I think replacing all of them in a single change would be cleaner.

Reply:
Yeah, fine by me to have a separate PR.